build-prybar.nix

```
@@ -21,11 +28,44 @@ in buildGoModule {
 
     vendorSha256 = null;
 
+    # This prebiuild hook will setup the compiler flags on demand based on the package
+    # If a language requires this, it MUST expost a ${pkgName}.pc file in its `lib/pkg-config`
+    # for pkg-config to detect it. Otherwise, we can't find the required flags.
     preBuild = ''
+        export CGO_LDFLAGS_ALLOW="-Wl,--compress-debug-sections=zlib"
+        ${if setFlags then ''
+            NIX_CFLAGS_COMPILE="$(pkg-config --cflags ${pkgName}) $NIX_CFLAGS_COMPILE"
+            NIX_LDFLAGS="$(pkg-config --libs-only-L  ${pkgName}) $(pkg-config --libs-only-l  ${pkgName}) $NIX_LDFLAGS"
```

```suggestion
            export NIX_CFLAGS_COMPILE="$(pkg-config --cflags ${pkgName}) $NIX_CFLAGS_COMPILE"
            export NIX_LDFLAGS="$(pkg-config --libs-only-L  ${pkgName}) $(pkg-config --libs-only-l  ${pkgName}) $NIX_LDFLAGS"
```
Not sure if this matters though

---

It shouldn't in this context since this is treated as a bash script with those variables already added I think, but let's see if adding it changes anything.